File: python/baekjoon/11724.py
# Created by Jello on 2017. 7. 5.
import queue

V, E = map(int, str(input()).split(' '))
vertices = [[] for _ in range(V+1)]
visit = set()
bfs_queue = queue.Queue()
cnt = 0

for i in range(E):
    v1, v2 = map(int, str(input()).split(' '))
    vertices[v1].append(v2)
    vertices[v2].append(v1)

for i in range(1, len(vertices)):
    if i not in visit:
        bfs_queue.put(i)
        visit.add(i)
        cnt += 1
        while bfs_queue.qsize() > 0:
            e = bfs_queue.get()
            for j in vertices[e]:
                if j not in visit:
                    visit.add(j)
                    bfs_queue.put(j)
                    print('put {}'.format(j))

print(cnt)

File: c++/baekjoon/11724.cpp
#include <iostream>
#include <queue>
#include <vector>
#include <list>
#include <set>
using namespace std;

int main(){
    int V, E, cnt=0;
    queue<int> bfs_queue;
    set<int> visit;


    cin >> V >> E;
    vector< list<int> > vertices(V+1);
    for(int i=1;i<=V;i++){
        list<int> l;
        vertices[i] = l;
    }

    for(int i=0;i<E;i++){
        int v1 = 0, v2 = 0;
        cin >> v1 >> v2;
        vertices[v1].push_back(v2);
        vertices[v2].push_back(v1);
    }

    for(int i=1;i<=V;i++){
        if(visit.find(i) == visit.end()){
            visit.insert(i);
            bfs_queue.push(i);
            cnt++;

            while(bfs_queue.size() > 0){
                int e = bfs_queue.front();
                bfs_queue.pop();

                for(list<int>::iterator j=vertices[e].begin();j!=vertices[e].end();j++){
                    if(visit.find(*j) == visit.end()){
                        bfs_queue.push(*j);
                        visit.insert(*j);
                    }
                }
            }
        }
    }

    cout << cnt;
}